src/main/java/checklst/task/Deadline.java

```diff
@@ -0,0 +1,58 @@
+package checklst.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import checklst.exception.ChecklstException;
+
+/**
+ * The Deadline class represents a Task with a specific Deadline.
+ */
+public class Deadline extends Task {
+
+    protected final LocalDate dueDate;
+
+    protected Deadline(String name, LocalDate dueDate) {
+        super(name);
+        this.dueDate = dueDate;
+    }
+
+    protected Deadline(String name, boolean completed, LocalDate dueDate) {
```

Comment:
Maybe boolean could be named isCompleted instead to sound more like a boolean?

Reply:
Thanks for the suggestion! Changed accordingly.